docs/3.0.0-beta.x/migration-guide/migration-guide-beta.20-to-3.0.0.md

```diff
@@ -0,0 +1,245 @@
+:::
+
+**Functions**
+
+You can leave your functions as is, we didn't change how they work.
+
+**Policies**
+
+You can leave your policies as is, we didn't change how they work.
+
+**Custom**
+
+Any custom configuration you have can still be used. You can read the [configuration documentation](../concepts/configurations.md) to know more.
+
+**Plugin**
+
+From now on, you can set your plugin configurations in `./config/plugins.js` or `./config/env/{env}/plugin.js`.
+
+**Example**
+
+```js
+module.exports = {
+  graphql: {
+    depthLimit: 5,
+  },
+};
+```
+
+## Database lifecycles
+
+We have replaced the old lifecycles that add a lot of issues with a new simpler lifecycle layer.
+
+You can read more [here](../concepts/models.md#lifecycle-hooks).
+
+## Email plugin settings
+
+Email plugin settings have been moved to files. Now you can configure your email provider directly in files.
+
+You can read the documentation [here](../plugins/email.md#configure-the-plugin) to update.
+
+## GraphQL changes
+
+If you are using the graphql `register` mutation, the input and response types have changed. You can check the code [here](https://github.com/strapi/strapi/pull/6047).
+
+## Remove `idAttribute` and `idAttributeType` options.
+
+Currently using the idAttribute and idAttributeType options can break strapi in many ways. Fixing this is going to require a lot of work on the database and content management layer.
+
+In an effort to make strapi more stable we have decided to remove those broken options for the time being. For users who want unique uuid fields for examples we recommend you create a uuid attribute and use the lifecycles function to populate it.
+
+## Proxy configuration
+
+In order to support hosting strapi with more flexibility, we have changed the way to configure the server proxy options and the admin panel path.
+
+### Proxy
+
+We replaced the `proxy` option found in `./config/server.json` by the `url` option.
+
+This option also makes the `admin.build.backend` option obsolete.
+
+This option tells strapi where it is hosted and is usefull for generating links or telling the admin panel where the API is available.
+
+**Before**
+
+**Path —** `./config/server.json`
+
+```json
+{
+  "proxy": {
+    "enabled": true,
+    "ssl": true,
+    "host": "domain.com",
+    "port": "1337"
+  }
+}
+```
+
+**After**
+
+**Path —** `./config/server.js`
+
+```js
+module.exports = {
+  //...
+  url: `https://domain.com:1337`,
+};
+```
+
+What you can now do is add a path to the url to host strapi in a sub path of your domain.
+
+```js
+module.exports = {
+  //...
+  url: `https://domain.com:1337/my-strapi-api`,
+};
+```
+
+::: warning
+Adding a sub path to the url doesn't mean your api is going to be prefixed. You will need to host your app behind a proxy and remove the prefix so strapi receives request like if they where made on the root `/` path.
+:::
+
+You can see this option in action in the following [deployment guide](../deployment/nginx-proxy.md).
```

Comment:
Change this to point to the higher level guide as there is more options than just Nginx (Caddy, HAProxy, and soon I'll have a guide for Apache)